### PR TITLE
Added HD-CzTorrent definition

### PR DESCRIPTION
**Pull request by mazmar**
My first fully create definiton. based on general template. I didn't find any mention of framework used on the site and the site itself is a mess ... 

for some reason there is some giant <if_is_displayed_2> code with <else_is_displayed_2> inside. no idea how that is supposed to work i don't work with html or javascript ...

every torrent is doubled on the site due the the fact written above but it was not possible to add "after" in rows definiton as the torrent is doubled as whole table not as a row...

Tested downloading and searching looks fine

### CODE REVIEW

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+description: "HD CzTorrent is a CZECH semi private site for TV / MOVIES / GENERAL"
+language: cs-cz
+type: semi-private
+encoding: windows-1250
+links:
+  - http://www.hd-cztorrent.cz/
+
+caps:
+  categorymappings:
+    - {id: 37, cat: Movies/DVD, desc: "DVD", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:13]:
```suggestion
    - {id: 37, cat: Movies/DVD, desc: "DVD"}
```
repeat for the rest

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  - name: username
+    type: text
+    label: Username
+
+  - name: password
+    type: password
+    label: Password
+
+login:
+  # use simple post login
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:72]: all comments can be removed

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  method: post
+  # target of the POST request
+  path: index.php?page=login
+  # list of POST parameters
+  inputs:
+    # using configured username and password from the prior settings block
+    uid: "{{ .Config.username }}"
+    pwd: "{{ .Config.password }}"
+  error:
+    - selector: tbody:has(td.lista > span:contains("Nesprávné uživatelské jméno"))
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:82]: also add login test, e.g.:
```
  test:
    path: index.php
    selector: a[href="logout.php"]
```
[generic example]

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  inputs:
+    page: "torrents"
+    search: "{{ .Keywords }}"
+    category: 0
+    uploader: 0
+    options: 0
+    active: 1
+    gold: 0
+  preprocessingfilters:
+   - name: replace
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:97]:
```suggestion
    - name: replace
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+     args: ["else:","else_"]
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:98]:
```suggestion
      args: ["else:", "else_"]
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      selector: a[href^="index.php?page=torrent-details"]
+      attribute: href
+    download:
+      selector: a[href^="download.php"]
+      attribute: href
+    poster:
+      selector: a[href^="index.php?page=torrent-details"]
+      attribute: onmouseover
+      filters:
+       - name: replace
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:122]:
```suggestion
        - name: replace
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+         args: "return overlib\\('<img src=(.+?) "
+    date:
+      selector: td:nth-child(6)
+      filters:
+       - name: dateparse
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:127]:
```suggestion
        - name: dateparse
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+         args: "02/01/2006"
+    size:
+      selector: td:nth-child(11)
+    seeders:
+      selector: td:nth-child(8) > a
+    leechers:
+      selector: td:nth-child(9) > a
+
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:135]:
```suggestion
```

**Review by ilike2burnthing — CHANGES_REQUESTED** [review]
Thanks for the contribution. Please see the requested changes.

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+      attribute: href
+    download:
+      selector: a[href^="download.php"]
+      attribute: href
+    poster:
+      selector: a[href^="index.php?page=torrent-details"]
+      attribute: onmouseover
+      filters:
+       - name: replace
+         args: "return overlib\\('<img src=(.+?) "
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:123]:
```suggestion
          args: "return overlib\\('<img src=(.+?) "
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml (hunk 3 above)
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:82]: there are other error messages that need to be trapped, not just invalid username:
Musíte zadat uživatelské jméno!
Nesprávné heslo

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 10]
```diff
@@ -0,0 +1,135 @@
+---
+id: htcztorrent
+name: HD-CzTorrent
+description: "HD CzTorrent is a CZECH semi private site for TV / MOVIES / GENERAL"
+language: cs-cz
+type: semi-private
+encoding: windows-1250
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:7]:
```suggestion
encoding: UTF-8
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+search:
+  paths:
+    - path: index.php
+  inputs:
+    page: "torrents"
+    search: "{{ .Keywords }}"
+    category: 0
+    uploader: 0
+    options: 0
+    active: 1
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:94]:
```suggestion
    active: 0
```
please support ALL torrents not just active.

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+    - selector: tbody:has(td.lista > span:contains("Nesprávné uživatelské jméno"))
+
+
+search:
+  paths:
+    - path: index.php
+  inputs:
+    page: "torrents"
+    search: "{{ .Keywords }}"
+    category: 0
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:91]:
```suggestion
    category: "{{ range .Categories }}{{.}};{{end}}"
```
please support multi-category searching

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+    uploader: 0
+    options: 0
+    active: 1
+    gold: 0
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:95]:
```suggestion
    gold: "{{ if .Config.freeleech }}3{{ else }}0{{ end }}"
```
Please support freeleech searching
see https://github.com/Jackett/Jackett/blob/6a213c6eabd0264392c48564ff16a9035fd13ca4/src/Jackett.Common/Definitions/greekdiamond.yml for corresponding config setting support.

**Review by garfield69 — CHANGES_REQUESTED** [review]
please add HD CzTorrent to the semi-private list section of the readme.

Also the site supports torrent sorting requests, so please take a look at https://github.com/Jackett/Jackett/blob/6a213c6eabd0264392c48564ff16a9035fd13ca4/src/Jackett.Common/Definitions/greekdiamond.yml which uses a similar site engine for the settings and search options for supporting sort.

Thanks.

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+name: HD-CzTorrent
+description: "HD CzTorrent is a CZECH semi private site for TV / MOVIES / GENERAL"
+language: cs-cz
+type: semi-private
+encoding: windows-1250
+links:
+  - http://www.hd-cztorrent.cz/
+
+caps:
+  categorymappings:
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:12]:
```suggestion
  categorymappings:
    # Film
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    - {id: 110, cat: Movies/3D, desc: "3D ORIGINÁL", default: false}
+    - {id: 111, cat: Movies, desc: "FILMY CZ TVORBA", default: false}
+    - {id: 98, cat: Movies/UHD, desc: "ORIGINAL JAZYK 4K", default: false}
+    - {id: 77, cat: Movies/UHD, desc: "ULTRA HD  4K", default: false}
+    - {id: 89, cat: Movies/HD, desc: "HD720 titulky ", default: false}
+    - {id: 93, cat: Movies/HD, desc: "HD 1080 titulky", default: false}
+    - {id: 107, cat: Movies, desc: "CAMERA RIP ", default: false}
+    - {id: 94, cat: Movies, desc: "h 265", default: false}
+    - {id: 81, cat: Movies, desc: "ORIGINAL JAZYK", default: false}
+    - {id: 95, cat: Movies/Other, desc: "FILM RETRO", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:37]:
```suggestion
    - {id: 95, cat: Movies/Other, desc: "FILM RETRO"}
    # HUDBA
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    - {id: 57, cat: Audio/Lossless, desc: "Hudba flac", default: false}
+    - {id: 58, cat: Audio/MP3, desc: "Hudba mp3", default: false}
+    - {id: 96, cat: Audio/Other, desc: "RETRO HUDBA", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:40]:
```suggestion
    - {id: 96, cat: Audio/Other, desc: "RETRO HUDBA"}
    # ostatni
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+    - {id: 64, cat: Books, desc: "E-books, PDF fotky", default: false}
+    - {id: 65, cat: Audio/Audiobook, desc: "Mluvene slovo", default: false}
+    - {id: 113, cat: Books/Other, desc: "obrazky fotky", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:43]:
```suggestion
    - {id: 113, cat: Books/Other, desc: "obrazky fotky"}
    # PORNO
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+    - {id: 11, cat: XXX, desc: "XXX 18+KLASIKA", default: false}
+    - {id: 99, cat: XXX, desc: "XXX GAY", default: false}
+    - {id: 100, cat: XXX, desc: "XXX ASIAN", default: false}
+    - {id: 101, cat: XXX, desc: "XXX 3D", default: false}
+    - {id: 106, cat: XXX/ImageSet, desc: "XXX-OBRÁZKY", default: false}
+    - {id: 108, cat: XXX, desc: "PORNO BDSM", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:49]:
```suggestion
    - {id: 108, cat: XXX, desc: "PORNO BDSM"}
    # HRY
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+    - {id: 73, cat: PC/Games, desc: "Hry, Games", default: false}
+    - {id: 97, cat: Console/PS4, desc: "HRY PS4", default: false}
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:51]:
```suggestion
    - {id: 97, cat: Console/PS4, desc: "HRY PS4"}
    # APLIKACE
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+    tv-search: [q, season, ep]
+    movie-search: [q]
+    music-search: [q]
+    book-search: [q]
+
+settings:
+  - name: username
+    type: text
+    label: Username
+
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:66]:
```suggestion
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+  - name: password
+    type: password
+    label: Password
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:69]:
```suggestion
    label: Password
  - name: freeleech
    type: checkbox
    label: Search freeleech only
    default: false
  - name: sort
    type: select
    label: Sort requested from site
    default: 3
    options:
      3: created
      5: seeders
      4: size
      2: title
  - name: type
    type: select
    label: Order requested from site
    default: 2
    options:
      2: desc
      1: asc
  - name: info
    type: info
    label: Results Per Page
    default: For best results, change the <b>Torrents per page:</b> setting to <b>100</b> on your account profile.
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+  inputs:
+    # using configured username and password from the prior settings block
+    uid: "{{ .Config.username }}"
+    pwd: "{{ .Config.password }}"
+  error:
+    - selector: tbody:has(td.lista > span:contains("Nesprávné uživatelské jméno"))
+
+
+search:
+  paths:
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:86]:
```suggestion
  paths:
    # http://www.hd-cztorrent.cz/index.php?page=torrents&search=the+avengers&category=0&uploader=0&options=0&active=0&gold=0&order=3&by=2
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml (hunk 12 above)
Comment (ilike2burnthing) [reply src/Jackett.Common/Definitions/hdcztorrent.yml:91]: @garfield69 any benefit to `if else` for this?
```suggestion
    category: "{{ if .Categories }}{{ range .Categories }}{{.}};{{end}}{{ else }}0{{ end }}"
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 23]
```diff
@@ -0,0 +1,135 @@
...
+    - path: index.php
+  inputs:
+    page: "torrents"
+    search: "{{ .Keywords }}"
+    category: 0
+    uploader: 0
+    options: 0
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:93]:
```suggestion
    # 0 filename, 1 file & desc, 2 description
    options: 0
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml (hunk 11 above)
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:94]:
```suggestion
    # 0 filename, 1 file & desc, 2 description
    active: 0
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml (hunk 13 above)
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:95]:
```suggestion
    # 0 all, 1 classic, 2 silver, 3 gold, 4 silver & gold
    gold: "{{ if .Config.freeleech }}3{{ else }}0{{ end }}"
    order: "{{ .Config.sort }}"
    by: "{{ .Config.type }}"
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 24]
```diff
@@ -0,0 +1,135 @@
...
+    active: 1
+    gold: 0
+  preprocessingfilters:
+   - name: replace
+     args: ["else:","else_"]
+
+  rows:
+    selector: else_is_displayed_2 tr:has(> td > a[href^=download])
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:101]:
```suggestion
    selector: table.lista > tbody > tr:has(a[href*=bookmark])
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 25]
```diff
@@ -0,0 +1,135 @@
...
+         args: "return overlib\\('<img src=(.+?) "
+    date:
+      selector: td:nth-child(6)
+      filters:
+       - name: dateparse
+         args: "02/01/2006"
+    size:
+      selector: td:nth-child(11)
+    seeders:
+      selector: td:nth-child(8) > a
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:132]:
```suggestion
      selector: td:nth-child(8)
```

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 26]
```diff
@@ -0,0 +1,135 @@
...
+    leechers:
+      selector: td:nth-child(9) > a
```
Comment (ilike2burnthing) [inline src/Jackett.Common/Definitions/hdcztorrent.yml:134]:
```suggestion
      selector: td:nth-child(9)
```

**Review by ilike2burnthing — CHANGES_REQUESTED** [review]
Made an account and played about with this, so there's a few more changes needed. You can just click `Commit suggestion` on everything you're ok with, rather than having to do it manually.

**Review by garfield69 — APPROVED** [review]
I went ahead and fixed up a few things and tweaked a few others.
Tested OK.
Thanks for contributing.

File: src/Jackett.Common/Definitions/hdcztorrent.yml [hunk 27]
```diff
@@ -123,8 +123,8 @@ search:
     uploader: 0
     # 0 filename, 1 file & desc, 2 description
     options: 0
-    # 0 all, 1 oactive only, 2 dead only
-    active: 1
+    # 0 all, 1 active only, 2 dead only
+    active: 0
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/hdcztorrent.yml]: oops, accidentally overrode the fix you snuck in while I was editing. fixed later with 7d9eea0

**Comment by garfield69** [comment]
v0.18.145

**Comment by mazmar** [comment]
Thanks for the suggestions and edits, I was away for a weekend

**Comment by karelkryda** [comment]
@gitthangbaby I have the same issue, did you solved it?

**Comment by gitthangbaby** [comment]
no, i gave up on CZ/SK trackers, but i will report once i reinstate the server and retest

**Pull request merged**